recipes/yaclib/all/conanfile.py

```diff
@@ -0,0 +1,129 @@
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain
+from conan.tools.scm import Version
+from conan.tools.files import copy, get, export_conandata_patches, apply_conandata_patches
+from conan.tools.layout import cmake_layout
+from conan.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.52.0"
+
+
+class YACLibConan(ConanFile):
+    name = "yaclib"
+    description = "lightweight C++ library for concurrent and parallel task execution"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/YACLib/YACLib"
+    license = "MIT"
+    topics = "c++", "async", "parallel", "concurrency"
+    settings = "os", "arch", "compiler", "build_type"
+
+    _yaclib_flags = {
+        "warn": [True, False],
+        "asan": [True, False],
+        "tsan": [True, False],
+        "ubsan": [True, False],
+        "lsan": [True, False],
+        "memsan": [True, False],
+        "coverage": [True, False],
+        "coro": [True, False],
+        "disable_futex": [True, False],
+        "disable_unsafe_futex": [True, False],
+        "disable_symmetric_transfer": [True, False],
+        "disable_final_suspend_transfer": [True, False],
+    }
+
+    options = {
+        "fPIC": [True, False],
+        "fault": [None, "thread", "fiber"],
+        **_yaclib_flags,
+    }
+
+    default_options = {
+        "fPIC": True,
+        "fault": None,
+        **{k: False for k in _yaclib_flags},
+    }
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "Visual Studio": "14.20",
+            "msvc": "192",
+            "clang": "8",
+            "apple-clang": "12",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if self.settings.compiler.get_safe("cppstd"):
+            tc.variables["YACLIB_CXX_STANDARD"] = self.settings.compiler.cppstd
+
+        flags = []
+        for flag in self._yaclib_flags:
+            if self.options.get_safe(flag):
+                flags.append(flag.upper())
+
+        if flags:
+            tc.variables["YACLIB_FLAGS"] = ";".join(flags)
+
+        tc.generate()
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, 17)
```

Comment:
If i understand the patches correctly

```suggestion
            check_min_cppstd(self, 20 if self.options.coro else 17)
```